### PR TITLE
F#src 3352 nvidia container fix

### PR DESCRIPTION
**Pull request by ToivoHn**

### CODE REVIEW

File: ansible/roles/products/common/docker-container/tasks/create-container/nvidia-with-setup-with-terminator.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      LD_LIBRARY_PATH: "/usr/lib/nvidia-418:{{ ansible_env.LD_LIBRARY_PATH }}"
+    network_mode: "host"
+    pid_mode: "host"
+    privileged: yes
+    interactive: yes
+    security_opts:
+    - "seccomp: unconfined"
+    volumes:
+    - /tmp/.X11-unix:/tmp/.X11-unix[rw]
+    - /usr/lib/nvidia-418:/usr/lib/nvidia-418
```
[the remark this answers is not in the document]
Comment (ToivoHn) [reply ansible/roles/products/common/docker-container/tasks/create-container/nvidia-with-setup-with-terminator.yml]: It's dynamic now in the latest commit

File: ansible/roles/installation/nvidia-docker/templates/override.j2 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#jinja2: trim_blocks:False
+[Service]
+ExecStart=
```
[the remark this answers is not in the document]
Comment (ToivoHn) [reply ansible/roles/installation/nvidia-docker/templates/override.j2]: No. I removed it in the latest commit

File: ansible/playbooks/teleop_deploy.yml [hunk 3]
```diff
@@ -1,8 +1,10 @@
 ---
-- name: Add IP address of all hosts to server (running Ansible)
-  hosts: server
+- name: Add IP address of all hosts to computer that is running Ansible
```
Comment (biofotis) [inline ansible/playbooks/teleop_deploy.yml]: Can you please make this separate PR?

Comment (ToivoHn) [reply ansible/playbooks/teleop_deploy.yml]: Done

Comment (ToivoHn) [reply ansible/playbooks/teleop_deploy.yml]: and I merged master into this

File: ansible/data/molecule/requirements.txt [hunk 4]
```diff
@@ -46,18 +48,21 @@ pycodestyle==2.5.0
...
 pyflakes==2.1.1
 PyNaCl==1.3.0
-pytest==4.4.1
+pyparsing==2.4.0
+pytest==4.6.3
 python-dateutil==2.8.0
 python-gilt==1.2.1
-PyYAML==3.13
-requests==2.21.0
+PyYAML==5.1.1
```
Comment (AndriyPt) [inline ansible/data/molecule/requirements.txt]: is it received from pip3 freeze after molecule installation on empty Ubuntu?

File: ansible/inventory/teleop/group_vars/client.yml [hunk 5]
```diff
@@ -3,7 +3,7 @@ image: "shadowrobot/shadow-teleop" # Docker image name
 tag: "kinetic-v0.0.11" # Tag of the exact Docker hub image to pull
 reinstall: false # Flag to know if the docker container should be fully reinstalled
 container_name: "teleop" # Name of the docker container
-nvidia_docker: 0 # Which version of nvidia-docker is enabled (0 means it is disabled)
+nvidia_docker: 2 # Which version of nvidia-docker is enabled (0 means it is disabled)
```
Comment (AndriyPt) [inline ansible/inventory/teleop/group_vars/client.yml]: please change it to 1 by default

File: ansible/inventory/teleop/group_vars/server.yml [hunk 6]
```diff
@@ -3,7 +3,7 @@ image: "shadowrobot/shadow-teleop" # Docker image name
 tag: "kinetic-v0.0.11" # Tag of the exact Docker hub image to pull
 reinstall: false # Flag to know if the docker container should be fully reinstalled
 container_name: "teleop" # Name of the docker container
-nvidia_docker: 0 # Which version of nvidia-docker is enabled (0 means it is disabled)
+nvidia_docker: 2 # Which version of nvidia-docker is enabled (0 means it is disabled)
```
Comment (AndriyPt) [inline ansible/inventory/teleop/group_vars/server.yml]: please change it to 1

File: ansible/roles/products/common/docker-container/tasks/main.yml [hunk 7]
```diff
@@ -4,15 +4,22 @@
   register: id
   changed_when: id.rc != 0
 
-- name: set docker_executable to be just docker if nvidia_docker is 0 or 2 (nvidia docker 2 uses just docker)
-  when: nvidia_docker | int ==0 or nvidia_docker | int ==2
+- name: set docker_executable if nvidia_docker is 0
+  when: nvidia_docker | int ==0
```
Comment (AndriyPt) [inline ansible/roles/products/common/docker-container/tasks/main.yml]: spaces should be symetrical

File: ansible/roles/products/teleop/control_machine/deploy/tasks/main.yml [hunk 8]
```diff
@@ -1,7 +1,7 @@
 ---
 - name: Setup script name
   set_fact:
-    setup_script: control-machine-setup.sh
+    setup_script: control_machine-setup.sh
```
Comment (AndriyPt) [inline ansible/roles/products/teleop/control_machine/deploy/tasks/main.yml]: missed underscore

File: ansible/inventory/teleop/group_vars/client.yml (hunk 5 above)
Comment (biofotis) [reply ansible/inventory/teleop/group_vars/client.yml]: I think 1 is not tested yet

File: ansible/roles/products/teleop/control_machine/deploy/tasks/main.yml [hunk 9]
```diff
@@ -76,13 +76,13 @@
...
   template:
     src: ../../../common/resources/templates/scripts/setup-hand.j2
     dest: "{{ shadow_hand_launcher_folder }}/{{ setup_script }}"
     mode: '755'
 
-- name: Copying control-machine setup script into docker container
+- name: Copying control_machine setup script into docker container
   changed_when: false
-  shell: "docker cp {{ shadow_hand_launcher_folder }}/control-machine-setup.sh {{ container_name }}:{{ setup_directory }}/{{ setup_script }}"
+  shell: "docker cp {{ shadow_hand_launcher_folder }}/control_machine-setup.sh {{ container_name }}:{{ setup_directory }}/{{ setup_script }}"
```
Comment (biofotis) [inline ansible/roles/products/teleop/control_machine/deploy/tasks/main.yml]: as above for undescore

File: ansible/data/molecule/requirements.txt (hunk 4 above)
Comment (ToivoHn) [reply ansible/data/molecule/requirements.txt]: yes. I created a new container based on empty Ubuntu and I ran "docker build --build-arg aurora_branch=F#SRC-3352_nvidia_container_fix xenial" for this Dockerfile: https://github.com/shadow-robot/aurora/blob/F%23SRC-3352_nvidia_container_fix/docker/molecule/base/xenial/Dockerfile and molecule was installed using requirements.txt in my branch, which includes PyYAML==5.1.1. After molecule installation is complete, pip3 freeze shows "PyYAML==5.1.1", so I know molecule installation has not downgraded PyYaml

File: ansible/inventory/teleop/group_vars/client.yml (hunk 5 above)
Comment (ToivoHn) [reply ansible/inventory/teleop/group_vars/client.yml]: Changed it to 1, will test with real hardware

File: ansible/inventory/teleop/group_vars/server.yml (hunk 6 above)
Comment (ToivoHn) [reply ansible/inventory/teleop/group_vars/server.yml]: Changed it to 1, will test with real hardware

File: ansible/roles/products/common/docker-container/tasks/main.yml (hunk 7 above)
Comment (ToivoHn) [reply ansible/roles/products/common/docker-container/tasks/main.yml]: Fixed in the latest commit
